submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/wiremock/FireMockDatabase.kt

```diff
@@ -47,6 +47,15 @@ class FireMockDatabase(
         }
     }
 
+    fun delete(fireOid: String) {
+        val record = recordsById.getValue(fireOid)
+        if (record.fileSystemPath != null) {
```

Comment:
we should not call delete on a file that does not have path? does we? if is not a valid scenario lests not handle issue in mock but rather let it fail so we identify any incosistency in the backend

Reply:
you're right, we in the system won't (or at least shouldn't) call delete over files without a path. However, this is a mock implementation of FIRE where you actually can call delete over any file regardless of whether it has a path or not, and isn't that the purpose of a mock? being as close as possible to the real thing? 🤔

Reply:
I see your point and agree, hard to balance beetween real behavior and error detection, what I was trying to said is that is good that mock fail if we call delete on file without path. It will helps us to detect an unwanted behavior but yes is not real.